### ISSUE TITLE
Modal positioning

### ISSUE THREAD

**Issue by cooper-joe** [dhis2/ui-core#576]
During the demo of the new data visualizer options it was clear that the height of the modal changing between tabs was not a user friendly experience. I've been thinking about how this can be avoided and what the component itself can do to help minimize this issue.

Current solution:
- Set a max height on the modal. This works, but it requires setting an arbitrary value that may or may not make sense, depending on how good of a guess one can make about the content.

Possible solution:
- I do think we should allow for different positioning of the modal. Allowing a 'top-aligned' modal with a standard offset from the top of the viewport would mean that only the action buttons would move around when the height of the modal changes. This would be a much less jarring experience than a vertically centered modal changing height.

Overly complex, uncessary solution:
- I do not think we should try to do any kind of height detection between tabs. This is overly complex and doesn't belong in the ui-core component.

So, in conclusion, I think we should add the following functionality to the modal component:
- Position: centered, top
- if `position = top`, offset = x (x defaults to 72px)

---

I think this will allow for the flexibility we need. Simpler, single page/tab modal views can continue to use centered vertical alignment. Complex, tabbed modals, or modals that display user generated data that prevents a reasonable max-height being set, can use top alignment.

What do you think @dhis2/team-apps?

**Comment by edoardo** [comment]
Perhaps using a viewport based height can be a solution for having a fixed size modal.
The downside is when you have little content and a big screen...
In any case it should be possible to override the `max-height` in the app.

**Comment by ismay** [comment]
> I do think we should allow for different positioning of the modal. Allowing a 'top-aligned' modal with a standard offset from the top of the viewport would mean that only the action buttons would move around when the height of the modal changes. This would be a much less jarring experience than a vertically centered modal changing height.

@cooper-joe That seems like a nice solution. One thing I would miss from the vertically centered dialog is that it seems more natural to have it displayed in the middle of the viewport. Especially on a large screen. We're all in agreement that the vertically centered one wasn't as nice? The height changing never really bothered me that much, maybe it's just something to get used to?

**Comment by Mohammer5** [comment]
Yeah, I'd prefer the top aligned over static/relative-to-body height as well, with 3 options in total:
1. `alignTop` (default)
1. `alignCenter`
1. `alignBottom`
